pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -371,9 +370,13 @@ func updateCEPUID(scopedLog *logrus.Entry, e *endpoint.Endpoint, localCEP *ciliu
 	if podHostIP == "" {
 		return fmt.Errorf("endpoint sync cannot take ownership of CEP: no pod HostIP")
 	}
-	if podHostIP != nodeIP {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that is not local: CEP's pod %q, pod's hostIP %q, cilium nodeIP %q)",
-			e.GetK8sPodName(), podHostIP, nodeIP)
+	if nodeIP := node.GetCiliumEndpointNodeIP(); podHostIP != nodeIP {
+		// Also checking node ipv6 for k8s dual stack with ipv6 preference where
+		// podHostIP is gonna be node ipv6
+		if nodeIPV6 := node.GetIPv6().String(); podHostIP != nodeIPV6 {
```

Comment:
What happens if the node doesn't have a v6 IP?

Reply:
@squeed